src/Server/RestRouterHandlers/SQLAnalyzerRestRouterHandler.cpp

```diff
@@ -229,16 +230,18 @@ std::pair<String, Int32> SQLAnalyzerRestRouterHandler::executePost(const Poco::J
 
         /// FIXME: CREATE TABLE ... AS SELECT ...
         /// FIXME: INSERT INTO TABLE ... SELECT ...
+        bool hasAggr = false;
         if (ast->as<ASTSelectWithUnionQuery>())
         {
             /// Interpreter will trigger ast analysis. One side effect is collecting
             /// required columns during the analysis process
             InterpreterSelectWithUnionQuery interpreter(ast, query_context, SelectQueryOptions());
+            hasAggr = interpreter.hasAggregation();
             block = interpreter.getSampleBlock();
         }
 
         auto query_type = queryType(ast);
-        return {buildResponse(query, rewritten_query, query_type, profile, block, query_context->requiredColumns()), HTTPResponse::HTTP_OK};
+        return {buildResponse(query, rewritten_query, query_type, profile, block, hasAggr, query_context->requiredColumns()), HTTPResponse::HTTP_OK};
```

Comment:
Let's clean the `has_aggr` and the corresponding logic in `QueryProfileVisitor` and `QueryProfileMatcher`